core/src/main/scala/org/broadinstitute/workbench/hamm/db/JobCostDAO.scala

```diff
@@ -79,10 +98,57 @@ object JobCostDAO {
       fr"FROM JOB_COST WHERE WORKFLOW_ID = ${callUniquekey.workflowId} AND" ++
       callFqnFragment ++ fr"=${callUniquekey.callFqn} AND" ++
       attemptFragment ++ fr"=${callUniquekey.attempt}").query[JobCost]
+
+  def getCallCostSql2(callUniquekey: CallUniquekey)(implicit session: DBSession) = {
+    import Binders._
+    val e = JobCost.syntax("e")
+    withSQL {
+      select.from(JobCost as e)
+        .where.eq(e.workflowId, callUniquekey.workflowId)
+        .and.eq(e.callFqn, callUniquekey.callFqn)
+        .and.eq(e.attempt, callUniquekey.attempt)
+    }.map(JobCost(e.resultName)).single().apply()
+  }
+
 }
 
 final case class CallFqn(asString: String) extends AnyVal
 final case class CallUniquekey(workflowId: WorkflowId, callFqn: CallFqn, attempt: Short, jobIndexId: Int)
-final case class JobCost(workflowId: WorkflowId, callFqn: CallFqn, attempt: Short, jobIndexId: Int, gcpJobId: Option[String], startTime: Instant, endTime: Instant, cost: Double){
-  val uniqueKey = CallUniquekey(workflowId, callFqn, attempt, jobIndexId)
+final case class JobCost(workflowId: WorkflowId, callFqn: CallFqn, attempt: Short, jobIndex: Int, vendorJobId: Option[String], startTime: Instant, endTime: Instant, cost: Double){
+  val uniqueKey = CallUniquekey(workflowId, callFqn, attempt, jobIndex)
+}
+
+object JobCost extends SQLSyntaxSupport[JobCost] {
+  override def tableName: String = "JOB_COST"
+  import Binders._
+  def apply(e: ResultName[JobCost])(rs: WrappedResultSet): JobCost = JobCost(
```

Comment:
method for converting a result set to a JobCost object

Reply:
what happens if `e` doesn't have one of the field?